CRM/ManualDirectDebit/Upgrader.php

```diff
@@ -180,6 +187,25 @@ private function deleteMessageTemplates() {
       ]);
     }
   }
+  /**
+   * Installs scheduled job
+   */
+  private function createScheduledJob() {
+    $domainID = CRM_Core_Config::domainID();
+
+    $params = [
+      'name' => 'Send Direct Debit Payment Collection Reminders',
+      'description' => 'Send Direct Debit Payment Collection Reminders',
+      'api_entity' => 'ManualDirectDebit',
+      'api_action' => 'run',
+      'run_frequency' => 'Daily',
+      'domain_id' => $domainID,
+      'is_active' => '0',
```

Comment:
Are you sure the scheduled job should start inactive?

On that note, we also need to implement onDisable and onEnable for the scheduled job too. On each case, the 'is_active' flag should be set to true or false, correspondingly.

Reply:
We have already implemented onDisable() and onEnable() for the scheduled job. Look at the method 'alterCustomValues()'  in Upgrader.

Reply:
Oh, ok, got it now. Still, I think it is a problem it wasn't obvious that method also takes care of enabling/disabling the scheduled job. So this is my suggestion of how we can improve this:

1. Separate CRM_ManualDirectDebit_Upgrader::$customValues into different class attributes (eg. $customOptionValues, $scheduledJobs, $processorTyes, $optionGroups).
2. Be sure to add a doc-block to each new attribute. 
3. Rename `alterCustomValues()` to `alterEntitiesValues()`.
4. Refactor `alterEntitiesValues()` to something like:
```php
private function alterCustomEntityValues($action) {
  $entities = array_merge($this->optionGroups, $this->scheduledJobs, $this->processorTypes, $this->optionValues);

  foreach ($entities as $customEntityValue) {
    $this->alterEntity(
      $customEntityValue['entityType'],
      $customEntityValue['searchValue'],
      isset($customEntityValue['searchField']) ? $customEntityValue['searchField'] : 'name',
      isset($customEntityValue['optionGroup']) ? $customEntityValue['optionGroup'] : FALSE,
      $action
    );
  }
}
```